packages/itwinui-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -536,67 +555,96 @@ export const ComboBox = React.forwardRef(
-                  onChange={handleOnInput}
-                  selectTags={
-                    isMultipleEnabled(selected, multiple)
-                      ? selected.map((index) => {
-                          const item = optionsRef.current[index];
+            <>
+              <ComboBoxInput
+                value={inputValue}
+                disabled={inputProps?.disabled}
+                {...inputProps}
+                onChange={handleOnInput}
+                selectTags={
+                  isMultipleEnabled(selectedIndexes, multiple)
+                    ? (selectedIndexes
+                        ?.map((index) => {
+                          const item: SelectOption<T> | undefined =
+                            optionsRef.current[index];
+
+                          if (item == null) {
+                            return undefined;
+                          }
+
                           return (
                             <SelectTag key={item.label} label={item.label} />
                           );
                         })
-                      : undefined
-                  }
-                />
-              </>
-              <ComboBoxEndIcon
-                {...endIconProps}
-                disabled={inputProps?.disabled}
-                isOpen={isOpen}
+                        .filter(Boolean) as JSX.Element[])
+                    : undefined
+                }
               />
-
-              {multiple ? (
-                <AutoclearingHiddenLiveRegion text={liveRegionSelection} />
-              ) : null}
-            </ComboBoxInputContainer>
-            <ComboBoxMenu as='div' {...dropdownMenuProps}>
-              {filteredOptions.length > 0 && !enableVirtualization
-                ? filteredOptions.map(getMenuItem)
-                : emptyContent}
-            </ComboBoxMenu>
-          </ComboBoxStateContext.Provider>
-        </ComboBoxActionContext.Provider>
+            </>
+            <ComboBoxEndIcon
+              {...endIconProps}
+              disabled={inputProps?.disabled}
+              isOpen={isOpen}
+            />
+
+            {multiple ? (
+              <AutoclearingHiddenLiveRegion text={liveRegionSelection} />
+            ) : null}
+          </ComboBoxInputContainer>
+          <ComboBoxMenu as='div' {...dropdownMenuProps}>
+            {filteredOptions.length > 0 && !enableVirtualization
+              ? filteredOptions.map(getMenuItem)
+              : emptyContent}
+          </ComboBoxMenu>
+        </ComboBoxStateContext.Provider>
       </ComboBoxRefsContext.Provider>
     );
   },
 ) as <T>(
   props: ComboBoxProps<T> & { ref?: React.ForwardedRef<HTMLElement> },
 ) => JSX.Element;
+
+// ----------------------------------------------------------------------------
+
+/** Returns either `option.id` or derives a stable id using `idPrefix` and `option.label` (without whitespace) */
+const getOptionId = (option: SelectOption<unknown>, idPrefix: string) => {
```

Comment:
why is this randomly moved to the bottom now? move it back up to reduce the diff